socketio/src/client/builder.rs

```diff
@@ -172,6 +174,25 @@ impl ClientBuilder {
         self
     }
 
+    /// Sets data sent in the opening request.
+    /// # Example
+    /// ```rust
+    /// use rust_socketio::{ClientBuilder};
+    /// use serde_json::json;
+    ///
+    /// let socket = ClientBuilder::new("http://localhost:4204/")
+    ///     .namespace("/admin")
+    ///     .auth(json!({ "password": "1337" }).to_string())
+    ///     .on("error", |err, _| eprintln!("Error: {:#?}", err))
+    ///     .connect();
+    ///
+    /// ```
+    pub fn auth<T: Into<String>>(mut self, auth: T) -> Self {
```

Comment:
If using `serde_json::Value`, it makes sense to change the bound to `Into<serde_json::Value>` or even `TryInto<serde_json::Value>` if we want to allow a `&str` as input.